### PR TITLE
Fix ParseAndCheckDocument

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/5173

untitled

### CODE REVIEW

**Comment by cartermp** [comment]
@KevinRansom / @brettfo / @TIHan we should take this for 15.8

**Comment by vasily-kirichenko** [comment]
Please test it on large files having this setting set to various values

image

File: vsintegration/src/FSharp.Editor/LanguageService/FSharpCheckerExtensions.fs [hunk 1]
```diff
@@ -60,19 +54,19 @@ type FSharpChecker with
...
-                    | StillRunning worker ->
+                    | Some x -> async.Return (Some x)
+                    | None ->
                         async {
                             match allowStaleResults, checker.TryGetRecentCheckResultsForFile(filePath, options) with
                             | true, Some (parseResults, checkFileResults, _) ->
                                 return Some (parseResults, checkFileResults)
                             | _ ->
-                                return! worker
+                                return! parseAndCheckFile
```
Comment (dsyme) [inline vsintegration/src/FSharp.Editor/LanguageService/FSharpCheckerExtensions.fs]: Please make `parseAndCheckFile` a function  `parseAndCheckFile ()`  to make things more obvious

**Comment by cartermp** [comment]
Link #5172

**Pull request merged**